desktop/apps/auction2/actions.js

```diff
@@ -12,11 +42,112 @@ export function updateArtworks(artworks) {
   }
 }
 
-export function toggleListView(isListView) {
+export function updateAggregatedArtists(aggregatedArtists) {
   return {
-    type: TOGGLE_LIST_VIEW,
+    type: UPDATE_AGGREGATED_ARTISTS,
     payload: {
-      isListView
+      aggregatedArtists
+    }
+  }
+}
+
+export function updateAggregatedMediums(aggregatedMediums) {
+  return {
+    type: UPDATE_AGGREGATED_MEDIUMS,
+    payload: {
+      aggregatedMediums
+    }
+  }
+}
+
+export function updateArtistParams(artistId) {
+  return (dispatch) => {
+    dispatch(updateArtistId(artistId))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateArtistId(artistId) {
+  return {
+    type: UPDATE_ARTIST_ID,
+    payload: {
+      artistId
     }
   }
 }
+
+export function updateMediumParams(mediumId) {
+  return (dispatch) => {
+    dispatch(updateMediumId(mediumId))
+    dispatch(fetchArtworks())
+  }
+}
+
+export function updateMediumId(mediumId) {
```

Comment:
One of the things that functional programming models really lend themselves to is refactoring. Having a lot of little things that conform to a certain pattern is ok, because then, later, when the need arises, you can sweep through and lift them up a level into a helper of some sort that reduces boilerplate. But for first passes (and until the need arises) its best just to just write everything out and keep it as straight forward as possible while larger patterns are laid out and settled upon in the app.